src/main/java/org/ibp/api/brapi/v2/germplasm/GermplasmResourceBrapi.java

```diff
@@ -0,0 +1,162 @@
+
+		final List<Germplasm> germplasmList = new ArrayList<>();
+
+		if (resultPage.getPageResults() != null) {
+			final ModelMapper mapper = new ModelMapper();
+			for (final GermplasmDTO germplasmDTO : resultPage.getPageResults()) {
+				final Germplasm germplasm = mapper.map(germplasmDTO, Germplasm.class);
+				germplasm.setCommonCropName(crop);
+				germplasmList.add(germplasm);
+			}
+		}
+
+		final Result<Germplasm> results = new Result<Germplasm>().withData(germplasmList);
+		final Pagination pagination = new Pagination().withPageNumber(resultPage.getPageNumber()).withPageSize(resultPage.getPageSize())
+			.withTotalCount(resultPage.getTotalResults()).withTotalPages(resultPage.getTotalPages());
+
+		final Metadata metadata = new Metadata().withPagination(pagination);
+
+		final EntityListResponse<Germplasm> entityListResponse = new EntityListResponse<>(metadata, results);
+
+		return new ResponseEntity<>(entityListResponse, HttpStatus.OK);
+
+	}
+
+	private GermplasmSearchRequestDto getGermplasmSearchRequestDto(final String germplasmPUI, final String germplasmName,
+		final String accessionNumber, final String studyDbId, final String synonym, final String genus, final String parentDbId, final String progenyDbId) {
+		final GermplasmSearchRequestDto germplasmSearchRequestDTO = new GermplasmSearchRequestDto();
+		germplasmSearchRequestDTO.setPreferredName(germplasmName);
+		germplasmSearchRequestDTO.setStudyDbId(studyDbId);
+		if (synonym != null) {
+			germplasmSearchRequestDTO.setGermplasmNames(Lists.newArrayList(synonym));
+		}
+		germplasmSearchRequestDTO.setParentDbId(parentDbId);
+		germplasmSearchRequestDTO.setProgenyDbId(progenyDbId);
+		if (genus != null){
+			germplasmSearchRequestDTO.setGermplasmGenus(Lists.newArrayList(genus));
+		}
+		if (accessionNumber != null){
+			germplasmSearchRequestDTO.setAccessionNumbers(Lists.newArrayList(accessionNumber));
+		}
+
+		if (germplasmPUI != null) {
+			germplasmSearchRequestDTO.setGermplasmPUIs(Lists.newArrayList(germplasmPUI));
+		}
+		return germplasmSearchRequestDTO;
+	}
+
+	private PagedResult<GermplasmDTO> getGermplasmDTOPagedResult(final GermplasmSearchRequestDto germplasmSearchRequestDTO,
+		final Integer currentPage, final Integer pageSize) {
+		final Integer finalPageNumber = currentPage == null ? BrapiPagedResult.DEFAULT_PAGE_NUMBER : currentPage;
+		final Integer finalPageSize = pageSize == null ? BrapiPagedResult.DEFAULT_PAGE_SIZE : pageSize;
+		return new PaginatedSearch()
+			.executeBrapiSearch(finalPageNumber, finalPageSize,
+				new SearchSpec<GermplasmDTO>() {
+
+					@Override
+					public long getCount() {
+						return GermplasmResourceBrapi.this.germplasmService.countGermplasmDTOs(germplasmSearchRequestDTO);
+					}
+
+					@Override
+					public List<GermplasmDTO> getResults(final PagedResult<GermplasmDTO> pagedResult) {
+						return GermplasmResourceBrapi.this.germplasmService
+							.searchGermplasmDTO(germplasmSearchRequestDTO, finalPageNumber, finalPageSize);
```

Comment:
Use Pageable for consistency with other services that require paginated response.